doc/dev/adr/0023-property-support.md

```diff
@@ -0,0 +1,312 @@
+
+* The request payload is an instance of `TProp`.
+  * For statically itemized properties, the optional fields in the `TProp` request object contain values for any Properties that are to be written.
+  * For dynamically itemized properties, the keys in the `TProp` request map indicate which Properties to write, and the corresponding values are what is to be written.
+* The response payload is an instance of `TStat`.
+  * For statically itemized properties, each optional field in the `TStat` response object indicates the status of the write operation for the Property named by the field.
+  * For dynamically itemized properties, each value in the `TStat` response map indicates the status of the write operation for the Property named by the key.
+  * The semantics of `TStatVal` values are determined by the service that implements the property.
+
+The *read* action is performed by issuing a 'read' Command request.
+
+* The request payload is an instance of `TCtrl`.
+  * For statically itemized properties, the optional fields in the `TCtrl` request object are `true` for each field whose Property is to be read.
+  * For dynamically itemized properties, the elements in the `TCtrl` request array indicate the names of the Properties to read.
+* The response payload is an instance of `TProp`.
+  * For statically itemized properties, the optional fields in the `TProp` response object contain values for any Properties that have been read.
+  * For dynamically itemized properties, the keys in the `TProp` response map indicate which Properties have been read, and the corresponding values are what was read.
+
+The *observe* action is performed by issuing an 'observe' Command request.
+
+* The request payload is an instance of `TCtrl`.
+  * For statically itemized properties, the optional fields in the `TCtrl` request object are `true` for each field whose Property is to be added to the notify list.
+  * For dynamically itemized properties, the elements in the `TCtrl` request array indicate the names of the Properties to add to the notify list.
+  * The maintainer may keep a single notify list across all consumers, or it may keep a separate notify list for each consumer; in the latter case, the maintainer must be provided with a way to identify the consumer that requested the observation.
+* The response payload is an instance of `TStat`.
+  * For statically itemized properties, each optional field in the `TStat` response object indicates the status of the observe operation for the Property named by the field.
+  * For dynamically itemized properties, each value in the `TStat` response map indicates the status of the observe operation for the Property named by the key.
+  * The semantics of `TStatVal` values are determined by the service that implements the property.
+  * The maintainer's response may additionally provide status values for the observation state of properties other than those named in the 'observe' request.
```

Comment:
I really liked your idea that
> it seemed to me that there should be some mechanism to find which items are currently on the watch list, and rather than adding another Command for this, it can be done by sending an empty list of additions in a "watch" Command.

I think this concept would be worth calling out here!